### PR TITLE
fix: Fix sticky bit related bugs

### PR DESCRIPTION
**Pull request by xiaguan**
Fixed a bug related to the sticky bit. Added tests` test_create_dir` and `test_create_symlink` because this bug was caused by the improper creation of symlinks. No tests related to the sticky bit were included, as it is difficult to simulate user switching in Rust tests. Relevant tests will be covered in `fstest`.

### CODE REVIEW

File: src/async_fuse/memfs/s3_metadata.rs [hunk 1]
```diff
@@ -1168,6 +1170,7 @@ impl<S: S3BackEnd + Send + Sync + 'static> S3MetaData<S> {
     }
 
     /// Rename helper function to pre-check
+    #[allow(clippy::too_many_lines)]
```
Comment (rogercloud) [inline src/async_fuse/memfs/s3_metadata.rs:1173]: Do we have an issue to track this function refactor?

File: src/async_fuse/memfs/s3_metadata.rs [hunk 2]
```diff
@@ -1521,6 +1535,7 @@ impl<S: S3BackEnd + Send + Sync + 'static> S3MetaData<S> {
     }
 
     /// Helper function to remove node locally
+    #[allow(clippy::too_many_lines)]
```
Comment (rogercloud) [inline src/async_fuse/memfs/s3_metadata.rs:1538]: Same as above.

File: src/async_fuse/memfs/s3_metadata.rs [hunk 3]
```diff
@@ -550,11 +550,11 @@ impl<S: S3BackEnd + Sync + Send + 'static> MetaData for S3MetaData<S> {
             Some(ref path) => Some(path.as_ref()),
             None => None,
         };
-        let uid = param.uid;
-        let gid = param.gid;
+        let user_id = param.uid;
```
Comment (rogercloud) [inline src/async_fuse/memfs/s3_metadata.rs:553]: I think `uid` and `gid` is accurate in this case. The naming is a kind of convention.

**Comment by Reto911** [comment]
Done by another PR.

**Pull request closed without merging**